streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -758,7 +758,9 @@ private int commitInternal(final Collection<Task> tasks) {
                 }
             }
 
-            commitOffsetsOrTransaction(consumedOffsetsAndMetadataPerTask);
+            if (!consumedOffsetsAndMetadataPerTask.isEmpty()) {
+                commitOffsetsOrTransaction(consumedOffsetsAndMetadataPerTask);
+            }
```

Comment:
This is an actually bug fix: `consumedOffsetsAndMetadataPerTask` could be empty, if only standby tasks (but no active tasks) are assigned to a thread.

Reply:
If we only have `StandbyTasks` assigned, the `RecordCollector` would not be initialized and thus the KafkaProducer would not initialize transactions and hence the offset commit would fail as we cannot begin a new transaction.